### DISCUSSION TITLE
Project stage code posted

### DISCUSSION THREAD

**Discussion by JNing0** [QuickPay-Operational-Performance/Data-and-code#83] [General]
@vibhuti6 I posted two files under folder "codes": "Rcode_ProjStage_0926.R" and "Rcode_stageRegress.R".

The first code file, "Rcode_ProjStage_0926.R", computes the project stage as is defined [in Section 2.2. here](https://github.com/QuickPay-Operational-Performance/Data-and-code/blob/master/QP1/QuickPay%20Delay%20Analysis_1010.pdf).  The data file I had when working on this is "dataNew.csv", which does not have the quarterly delay data.  Here is the description of the data (you'll see this at the beginning of the code):
`# 309,143-by-15 data frame. Each row is a project`
`# col. 1:    project Key`
`# col. 2-13: last reported end date at the end of 12 quarters starting from 2009.12.31`
`# col. 14:   start date of each project`
`# col. 15:   whether the project is a small business`
If my recollection is correct, this file does not have project characteristics such as initial budget or budget changes. Those were in a separate file (or you said that the budget entries were incorrect in this file, don't remember exactly).

The second code file, "Rcode_stageRegress.R", uses the project stage computed in the first code file. It
-  discretizes the continuous project stage into three categories: Early Stage, Middle Stage, or Late Stage,
-  does descriptive statistics, and 
- runs the regressions.

Let me know if you'd like to walk you through the code. Thanks!

**Comment by vibhuti6** [comment]
Thanks, Jie. I will try to rerun this with the new dataset, and follow up if I have any questions.

**Comment by vibhuti6** [comment]
Hi Jie, a quick follow up on this. Do you recall which file you used as "dataNew.csv" in Rcode_ProjStage_0926.R? If you could point me to the dropbox link for the file, it will help me figure out what has changed and which is the latest file we should be using instead. That way, we (hopefully) can run the same code easily. Thank you!

**Reply by vibhuti6** [reply]
I may be mistaken but looking at the code, it appears that "dataNew.csv" is a file that you processed. If yes, could you please share the code for processing it, and the raw file (from dropbox) used for it? No rush as I am currently working on revising the draft but just wanted to post here so we remember. Thanks!

**Reply by JNing0** [reply]
You are right. It is a processed file. The processing code is [here](https://github.com/QuickPay-Operational-Performance/Data-and-code/blob/master/codes/Rcode_Data%20processing_0924.R).

I found this issue #72 that contains all data sets you shared with me. The project stage code worked with the first file you shared, which contains info over fiscal year 2010-2018. I took out data on the first QP implementation and made it "dataNew."

**Reply by JNing0** [reply]
To correct my earlier reply, I used the "qp_resampled_fy10_to_fy12", as shown in the data processing code I just uploaded. Thanks!

**Reply by vibhuti6** [reply]
Great, thanks! Here's the new resampled file: https://www.dropbox.com/s/cjt4wlukkqpdw2y/qp_resampled_data_fy10_to_fy12_with_zero_obs.csv?dl=0 

So essentially we need to run the same codes but use the above file instead. I am happy to do it after I finish the draft. Or if you would like to do it, that'd be great as well. Thanks!

**Comment by vibhuti6** [comment]
Hi Jie, could you please also share your code for time-varying treatment effects? I will incorporate it into mine (to have consistent table formats). We don't currently have these results in the paper but we do plan on including it. Thank you!

**Reply by JNing0** [reply]
Sure. That code is in [this file](https://github.com/QuickPay-Operational-Performance/Data-and-code/blob/master/codes/Rcode_nQrtrNew_1107.R), line 209-303. I only looked at nonzero delays. Let me know if you have any questions!
